cmake/Platform/Windows/Packaging/Tests/conftest.py

```diff
@@ -0,0 +1,141 @@
+        self.project_build_path = self.project_path / 'build/Windows'
+        self.project_bin_path = self.project_build_path / 'bin/profile'
+
+        # start a log reader thread to print the output to screen
+        self.log_reader = io.open(self.temp_file.name, 'r', buffering=1)
+        self.log_reader_thread = threading.Thread(target=self._tail_log, daemon=True)
+        self.log_reader_thread.start()
+        self.log_reader_shutdown = False
+    
+    def _tail_log(self):
+        """ Tail the log file and print to screen for easy viewing in Jenkins. """
+        while True:
+            line = self.log_reader.readline()
+            if line == '':
+                if self.log_reader_shutdown:
+                    return
+            else:
+                print(line, end='')
+
+    def _get_local_installer_path(self, uri):
+        """ Return the local path to the installer, downloading the remote file if necessary """
+        parsed_uri = urllib.parse.urlparse(uri)
+        if parsed_uri.scheme in ['http', 'https', 'ftp', 'ftps', 's3']:
+            installer_path = Path(self.temp_dir) / 'installer.exe'
+
+            if parsed_uri.scheme == 's3':
+                session = boto3.session.Session()
+                client = session.client('s3')
+                client.download_file(parsed_uri.netloc, parsed_uri.path.lstrip('/'), str(installer_path))
+            else:
+                urllib.request.urlretrieve(parsed_uri.geturl(), installer_path)
+
+            # verify with SignTool if available
+            self._verify(installer_path)
+
+            return installer_path
+        else:
+            # strip the leading slash from the file URI 
+            return Path(parsed_uri.path.lstrip('/')).resolve()
+
+    def _verify(self, file):
+        """ Assert if signtool.exe exists and cannot verify the given file. """
+        try:
+            signtool_path_result = run(['powershell','-Command',"(Resolve-Path \"C:\\Program Files (x86)\\Windows Kits\\10\\bin\\*\\x64\" | Select-Object -Last 1).Path"], text=True, capture_output=True)
+            if signtool_path_result.returncode == 0:
+                signtool_path = Path(signtool_path_result.stdout.strip()) / 'signtool.exe'
+                signtool_result = self.run([str(signtool_path),'verify', '/pa', str(file)])
+                assert signtool_result.returncode == 0
+        except FileNotFoundError as e:
+            pass
+
+    def run(self, command, timeout=None, cwd=None):
+        """ Run command with the correct environment and logging settings. """
+        self.temp_file.write('\n' + list2cmdline(command) + '\n')
+        self.temp_file.flush()
+        return run(command, timeout=timeout, cwd=cwd, stdout=self.temp_file, stderr=self.temp_file, text=True)
+
+    def cleanup(self):
+        """ Clean up temporary testing artifacts. """
+
+        # wait a few seconds for processes to stop using resources 
+        sleep(5)
```

Comment:
Umm, Sleeps are never solution to a problem, they exist to cover up an existing issue.

A better way is to check if a process is still running.

Reply:
I think I added this to give Asset Processor and friends time to shutdown, I'll try removing it to see if it is no longer needed.  There may not be an easy way to detect the processes holding onto the resources because of the number of processes (and child processes) in use.

Reply:
Just for clarity: As part of the workspace unmounting script (`incremental_build_util.py`), we sigkill all processes prior to the unmount using `psutil`. There may be enough time between this cleanup and the unmount to gracefully terminate.

Reply:
I removed the `sleep` and Jenkins was still green, so I'm not currently able to repro the original issue for which I added the `sleep`.  If it starts to repro again, I'll proceed with more in-depth process tracking and monitoring.